app/controllers/application_controller.rb

```diff
@@ -1,2 +1,12 @@
 class ApplicationController < ActionController::Base
+  before_action :require_login
```

Comment:
@JuanMaGiraldo how a **before_action** works?

Reply:
Before_action is a filter which the other controllers inherit by being defined in the ApplicationController (the other components inherit from it). This filter runs before of executing controller actions in response to route requests. In this case, this filter requires that the user be logged in to be able to execute actions defined in the other controllers.